src/components/Video.tsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import DownloadButton from './DownloadButton';
+import FavoriteButton from './FavoriteButton';
+import Label from './Label'
+import { get } from 'idb-keyval';
+import { videoStore } from '../index'
+import { Resource, videoOrArticle } from '../store/resourcesSlice'
+import { makeStyles, createStyles } from "@material-ui/core/styles"
+
+type VideoProps = {
+  resId: number,
+  videoDetails: Resource,
+}
+
+const styles = makeStyles(() => createStyles({
+  labelList: {
+    display: "flex",
+    flexDirection: "row",
+
+  }
+}))
+
+const Video = ({ resId, videoDetails }: VideoProps) => {
+
+  const classes = styles();
+
+  function getId(url) {
+    const regExp = /^.*(youtu.be\/|v\/|u\/\w\/|embed\/|watch\?v=|&v=)([^#&?]*).*/;
+    const match = url.match(regExp);
+    return (match && match[2].length === 11) ? match[2] : null;
+  }
+
+  let videoUrl;
+
+  if (videoDetails.isCached) {
+    get(resId, videoStore).then(videoBlob => {
+      if (videoBlob !== undefined) {
+        videoUrl = URL.createObjectURL(videoBlob);
```

Comment:
Is this a placeholder until WCS is able to fix their S3 settings? It seems like right now it's treating cached videos as the same as other URLs and trying to load them from Youtube